wire-runtime-swift/src/main/swift/propertyWrappers/CopyOnWrite.swift

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+
+/// A property wrapper which allows easily implementing Copy-on-Write for value types.
+@propertyWrapper
+public struct CopyOnWrite<Value> {
+
+    /// The underlying value represented by the property wrapper.
+    ///
+    /// When writing to the the property, the property wrapper will
+    /// perform copy-on-write by utilizing `isKnownUniquelyReferenced`
+    /// to make a copy of the underlying storage.
+    public var wrappedValue: Value {
+        get {
+            storage.value
+        }
+
+        set {
+            if isEqual(self.wrappedValue, newValue) {
```

Comment:
- [ ] remove this bit

Reply:
+1 feel to remove, this was largely an optimization where we saw most "theme setters" setting values to the same value. Assuming things are different with protos, so seems fine to remove.